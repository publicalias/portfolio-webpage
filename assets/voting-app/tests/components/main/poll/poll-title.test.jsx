"use strict";

//local imports

const PollTitle = require("../../../../scripts/components/main/poll/poll-title");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { initSchema, newUser } = require("schemas/master");
const { newPoll, newState } = require("schemas/voting-app");
const { mockResults, testMock } = require("test-helpers/meta-tests");
const { initTestSnapshot, reactTests } = require("test-helpers/react-tests");

//setup

beforeAll(reactTests.setup);

//poll title

describe("poll title", () => {

  const { testShallow } = testWrapper(PollTitle);

  const testSnapshot = initTestSnapshot(testShallow);

  const testProps = (role, data = {}) => {

    const newForm = initSchema(newState().form);

    const fn = (poll, user = {}) => {
      testSnapshot({ user }, {
        poll,
        role
      });
    };

    switch (role) {
      case "form":
        fn(newForm({ title: data.title }), newUser({ name: data.author }));
        break;
      case "view":
        fn(newPoll(data));
        break;
      default:
        fn(newPoll());
    }

  };

  it("should match snapshot (default)", () => testProps());

  it("should match snapshot (form)", () => testProps("form"));

  it("should match snapshot (form, title)", () => testProps("form", { title: "Title A" }));

  it("should match snapshot (form, author)", () => testProps("form", { author: "Author A" }));

  it("should match snapshot (view)", () => testProps("view"));

  it("should match snapshot (view, title)", () => testProps("view", { title: "Title A" }));

  it("should match snapshot (view, author)", () => testProps("view", { author: "Author A" }));

});

describe("poll title (input)", () => {

  const { testMount } = testWrapper(PollTitle);

  const testInput = (event, fn) => {

    const select = jest.fn(() => ({
      class: jest.fn(),
      text: jest.fn(() => "")
    }));

    PollTitle.injected.select = select;

    const { props, wrapper } = testMount(null, {
      poll: newPoll(),
      role: "form"
    });

    wrapper.find(".js-ref-input").simulate(event);

    fn(props, select);

    wrapper.unmount();

  };

  it("should call formSetTitle on blur", () => testInput("blur", (props, select) => {

    const { actions: { formSetTitle } } = props;

    const [a] = mockResults(select);

    testMock(select, [".js-ref-input"]);

    testMock(a.text, []);

    testMock(formSetTitle, [""]);

  }));

  it("should toggle is-hidden on input", () => testInput("input", (props, select) => {

    const [a, b] = mockResults(select);

    testMock(select, [".js-hide-placeholder"], [".js-ref-input"]);

    testMock(a.class, ["is-hidden", true, ""]);
    testMock(b.text, []);

  }));

});
