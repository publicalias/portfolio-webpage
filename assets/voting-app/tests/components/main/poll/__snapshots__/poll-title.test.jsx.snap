// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`poll title (form) should match snapshot (author) 1`] = `
<div
  className="u-align-center"
>
  <input
    className="c-poll-display__input qa-title-input u-margin-half"
    maxLength="100"
    onChange={[Function]}
    placeholder="Untitled"
    value=""
  />
  <h4>
    By Author A
  </h4>
</div>
`;

exports[`poll title (form) should match snapshot (default) 1`] = `
<div
  className="u-align-center"
>
  <input
    className="c-poll-display__input qa-title-input u-margin-half"
    maxLength="100"
    onChange={[Function]}
    placeholder="Untitled"
    value="Title A"
  />
  <h4>
    By Anonymous
  </h4>
</div>
`;

exports[`poll title (view) should match snapshot (author) 1`] = `
<div
  className="u-align-center"
>
  <h3
    className="u-margin-half"
  >
    Untitled
  </h3>
  <h4>
    By Author A
  </h4>
</div>
`;

exports[`poll title (view) should match snapshot (default) 1`] = `
<div
  className="u-align-center"
>
  <h3
    className="u-margin-half"
  >
    Untitled
  </h3>
  <h4>
    By Anonymous
  </h4>
</div>
`;

exports[`poll title (view) should match snapshot (title) 1`] = `
<div
  className="u-align-center"
>
  <h3
    className="u-margin-half"
  >
    Title A
  </h3>
  <h4>
    By Anonymous
  </h4>
</div>
`;
